src/Summoner/Template/Stack.hs

```diff
@@ -9,11 +9,13 @@ import NeatInterpolation (text)
 import Summoner.GhcVer (GhcVer (..), baseVer, latestLts, showGhcVer)
 import Summoner.Settings (Settings (..))
 import Summoner.Tree (TreeFs (..))
+import Summoner.Default (defaultGHC)
 
 
 stackFiles :: Settings -> [TreeFs]
-stackFiles Settings{..} = map createStackYaml settingsTestedVersions
+stackFiles Settings{..} = map createStackYaml ghcVersions
  where
+    ghcVersions = sortNub (defaultGHC : settingsTestedVersions)
```

Comment:
I think that one of the point of this refactoring is to remove explicit pattern-matching on `Ghc843` and compare GHC version with the `defaultGHC` in `ver` variable below.